ext/standard/basic_functions.c

```diff
@@ -4662,14 +4662,16 @@ PHP_FUNCTION(time_sleep_until)
 	/* 1sec = 1000000000 nanoseconds */
 	php_req.tv_nsec = (long) ((c_ts - php_req.tv_sec) * 1000000000.00);
 
-	while (nanosleep(&php_req, &php_rem)) {
-		if (errno == EINTR) {
+	do {
+		int rc = nanosleep(&php_req, &php_rem);
```

Comment:
Not sure I understand what the problem here was. If nanosleep returns -1 then we'll go into the while loop and handle errno == EINTR, no?

Reply:
You're totally right ...